### ISSUE TITLE
[HOLD] [Waiting on #19728]Web - Uploading attachment... text is still present when attachment upload results in error

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#22344]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to any chat
2. Click plus button and choose "Add attachment"
3. Choose unsupported file type like ".c", you can create it by creating file ending with ".c"

## Expected Result:
 If upload attachment fails and show an error, "Uploading attachment..." text should change to some text which correctly represents current state

## Actual Result:
 "Uploading attachment..." is still present, even the request fails and shows an error

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.37-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

![image (15)](https://github.com/Expensify/App/assets/93399543/c5c731ea-0fbf-4eb1-9588-c0c5f1de68a6)

https://github.com/Expensify/App/assets/93399543/dad6cc88-c3ec-4c01-b2c9-39a6b38f205e

![Screenshot 2023-07-06 at 11 21 13 AM](https://github.com/Expensify/App/assets/93399543/c716f87e-fc71-4a7e-ba74-f6ea06a7a23a)

**Expensify/Expensify Issue URL:**
**Issue reported by:** @alitoshmatov
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1688624919767169

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kbecciv** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
Uploading attachment... text is still present when attachment upload results in error
### What is the root cause of that problem?
When requests results in error we are not updating attachment report action
### What changes do you think we should make in order to solve the problem?
We should supply report action failure data here, covering this case:
https://github.com/Expensify/App/blob/3a6ec7044552f64f8d60dcf9439d3be3d4a697a8/src/libs/actions/Report.js#L297-L306
Something like this:
```javascript
value: _.mapObject(optimisticReportActions, (action) => ({
                ...action,
                errors: ErrorUtils.getMicroSecondOnyxError(‘report.genericAddCommentFailureMessage’),
                ...(file
                    ? {
                          message: [
                              {
                                  type: CONST.REPORT.MESSAGE.TYPE.COMMENT,
                                  html: ‘Failed attachment’,
                                  text: ‘Failed attachment’,
                              },
                          ],
                      }
                    : {}),
            })),
```
We can supply any desired message here, and write this code in better way
### What alternative solutions did you explore? (Optional)

**Comment by binary3oul** [comment]
I want to be a contributor.

**Comment by binary3oul** [comment]
Contributor details
Your Expensify account email: solarteacostaw@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01f25f9b8b73206482

**Comment by johncschuster** [comment]
Sweet! Thanks for the quick review!

**Comment by alitoshmatov** [comment]
I think it is not related to https://github.com/Expensify/App/issues/19728. Since this issue is not about backend resulting in error, but how app should handle this error, and change `Uploading attachment...` text when any error occurs.

**Comment by johncschuster** [comment]
Thanks for chiming in @alitoshmatov! @aimane-chnaif, what do you think?

**Comment by johncschuster** [comment]
I'm not following your train of thought, @aimane-chnaif. Can you clarify your comment? Does that mean you still feel this issue should be on hold while we wait for https://github.com/Expensify/App/issues/19728?

**Comment by johncschuster** [comment]
Bump!

**Comment by johncschuster** [comment]
Oh great! In that case, can we close this issue, or should we wait until the behavior is resolved?
